RohanVania/TestingInJavascript
# Completed Testing Basics module you can refer code for more understanding

In this module we learn about basics of Testing
Different types of testing
- Unit Testing
- Integration Testing 
- End to End Testing
- Manual Testing

We Learnt how to write our first test case different kind of function to use
We Learned the AAA Pattern Arrange, Act, Assert and why is it useful

How to catch error if the funtion does not catch on handle errors 
//* One Way to do this
```
export function add(numbers) {
  let sum =0
  for (const number of numbers) {
    sum += +number;
  }
  return sum;
}

test('Throw error if no values are passed',()=>{
    try{
        const result=add()
    }catch(err){
        expect(err).toBeDefined()
        
    }
})

//* Another way to do this 
test('Second way of handling Throw error if no values are passed',()=>{
    const result=()=>{
        add();
    }
    expect(result).toThrow();
})
```

How to use multiple testing in one function

We learned what are test suite and how we use it to organize our test cases.
